utils/post.js

```diff
@@ -310,39 +310,54 @@ function modifyReactionsPost(post, isAnonimous = true) {
   return newPost;
 }
 
-async function filterFeeds(userId, feeds = [], feed_id = null) {
-  const newResult = [];
+const isExpiredPost = (item) => {
+  const now = moment().valueOf();
+  const dateExpired = moment(item?.expired_at).valueOf();
 
-  for (const item of feeds || []) {
-    const now = moment().valueOf();
-    const dateExpired = moment(item?.expired_at).valueOf();
+  return dateExpired < now;
+};
 
-    if (dateExpired < now) {
-      if (feed_id) {
-        deleteActivityFromUserFeed('topic', feed_id, item.id);
-      }
-      continue;
-    }
-    let newItem = {...item};
+const deleteExpiredPostFromFeed = (item, feed_id) => {
+  if (feed_id) {
+    deleteActivityFromUserFeed('topic', feed_id, item.id);
+  }
+};
 
-    if (newItem.anonimity) {
-      newItem.actor = {};
-      newItem.to = [];
-      newItem.origin = null;
-      newItem.object = '';
-    }
+const modifyNewItemAnonymity = (newItem) => {
+  if (newItem.anonimity) {
+    newItem.actor = {};
+    newItem.to = [];
+    newItem.origin = null;
+    newItem.object = '';
+  }
+  return newItem;
+};
 
-    newItem = modifyReactionsPost(newItem, newItem.anonimity);
+async function filterFeeds(userId, feeds = [], feed_id = null, threshold = null) {
+  const results = await Promise.allSettled(
+    feeds.map(async (item) => {
+      const expired = isExpiredPost(item);
+      if (expired) {
+        deleteExpiredPostFromFeed(item, feed_id);
+        return null;
+      }
+      if (threshold && (item.final_score || 0) < threshold) {
+        return null;
+      }
+      let newItem = {...item};
 
-    const isValidPollPost = item.verb === POST_VERB_POLL && item?.polls?.length > 0;
+      newItem = modifyNewItemAnonymity(newItem);
+      newItem = modifyReactionsPost(newItem, newItem.anonimity);
 
-    if (isValidPollPost) newItem = await modifyPollPostObject(userId, newItem);
-    else newItem = await modifyPostLinkPost(DomainPage, newItem);
+      const isValidPollPost = item.verb === POST_VERB_POLL && item?.polls?.length > 0;
 
-    newResult.push(newItem);
-  }
+      if (isValidPollPost) newItem = await modifyPollPostObject(userId, newItem);
+      else newItem = await modifyPostLinkPost(DomainPage, newItem);
 
-  return newResult;
+      return newItem;
+    })
+  );
+  return results.filter((item) => item !== null);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `filterFeeds` function has been refactored to use `Promise.allSettled` which is a good improvement for performance as it allows all promises to be processed concurrently. However, there's no error handling for rejected promises. You should add logic to handle any promise rejections.
```diff
async function filterFeeds(userId, feeds = [], feed_id = null, threshold = null) {
  const results = await Promise.allSettled(
    feeds.map(async (item) => {
      // ... existing code ...
      return newItem;
    })
  );
  return results.filter((item) => item !== null && item.status === 'fulfilled').map(item => item.value);
}
```
In the above fix, I've added a check for `item.status === 'fulfilled'` to ensure we only include items that were successfully resolved. The `.map(item => item.value)` is used to extract the value of the fulfilled promises.


<!-- This is an auto-generated comment by OSS CodeRabbit -->